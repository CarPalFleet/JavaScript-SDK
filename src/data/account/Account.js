/**
 * @fileoverview This file contains all account related functions that are triggered by a User
 */

import axios from 'axios';
import endpoints from '../Endpoint';
import camelize from 'camelize';
import { apiResponseErrorHandler } from '../utility/Util';

/**
 * Request reset password token
 * @param {string} email
 * @return {object} Promise resolve/reject
 */
export const resetPasswordRequestAsync = async (email) => {
  try {
    const response = await axios({
      method: 'POST',
      url: endpoints.API_V3.REQUEST_PASSWORD_RESET,
      headers: { 'Content-Type': 'application/json' },
      data: {
        email,
      },
    });
    return camelize(response);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};

/**
 * Reset password with valid token
 * @param {string} token
 * @param {string} email
 * @param {string} password
 * @param {string} passwordConfirmation
 * @return {object} Promise resolve/reject
 */
export const resetPasswordAsync = async (
  token,
  email,
  password,
  passwordConfirmation
) => {
  try {
    const response = await axios({
      method: 'PUT',
      url: endpoints.API_V3.REQUEST_PASSWORD_RESET,
      headers: { 'Content-Type': 'application/json' },
      data: {
        token,
        email,
        password,
        passwordConfirmation,
      },
    });
    return camelize(response.data);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};

/**
 * Validate Reset Password Token
 * @param {string} token
 * @return {object} Promise resolve/reject
 */
export const validateResetPasswordTokenAsync = async (token) => {
  try {
    const response = await axios({
      method: 'GET',
      url: endpoints.API_V3.VALIDATE_PASSWORD_RESET_TOKEN.replace(
        '{token}',
        token
      ),
      headers: { 'Content-Type': 'application/json' },
      data: {
        token,
      },
    });
    return camelize(response.data.data);
  } catch (e) {
    return apiResponseErrorHandler(e);
  }
};
