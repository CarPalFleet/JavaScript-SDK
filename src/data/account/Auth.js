import axios from 'axios';
import endpoints, { cid, secret } from '../Endpoint';

export const getTokenAsync = async (email, password)=>{
    try{
        const response = await axios({method: 'post', 
                                      url: endpoints.OAUTH,
                                      headers: {'Content-Type': 'application/json'},
                                      data: {
                                        username: email,
                                        password,
                                        grant_type: "password",
                                        client_id: cid,
                                        client_secret: secret,
                                        scope: ""
                                      }})
        return response.data.data;     
    }catch(e){
        return null
    }
}            