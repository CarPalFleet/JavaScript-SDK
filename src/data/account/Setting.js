/**
 * @fileoverview This file contains all Settings related functions that are triggered by a User
 */

import axios from 'axios';
import endpoints from '../Endpoint';
import camelize from 'camelize';
import {apiResponseErrorHandler} from '../utility/Util';

/** Retrieve User's settings
 * There are 3 setting types in the setting table
 * 1. routing, 2. my-order, 3. driver-list
 * In routing type, it includes user time line setting (15 min, 30 min, 45 min etc.)
 * Retrieve table settings from my-order type OR driver-list.
 * @param {integer} userId
 * @param {string} type # routing, my-order, driver-list
 * @param {string} token
 * @return {object} promise (resolve/reject)
 */
export const getUserSettingsAsync = async (userId, type, token) => {
  try {
    const response = await axios({
      method: 'GET',
      url: `${endpoints.USER_SETTINGS.replace(
        '{0}',
        userId
      )}?type=${type}`,
      headers: {Authorization: token},
    });
    return camelize(response.data);
  } catch (e) {

    return apiResponseErrorHandler(e);
  }
};
