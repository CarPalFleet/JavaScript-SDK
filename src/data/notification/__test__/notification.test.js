import {
  getNotificationsAsync,
  deleteNotificationAsync,
  getUserNotificationsAsync,
  putUserNotification,
  deleteUserNotificationAsync,
  getUserNotificationCountAsync,
} from '../Notification';
import { getTokenAsync } from '../../account/Auth';
import CONFIG from './Config';

describe('Test for retrieving all notifications', () => {
  let token;
  beforeAll(async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
    token = await getTokenAsync(
      CONFIG.email,
      CONFIG.password,
      CONFIG.clientId,
      CONFIG.clientSecret
    );
  });

  it('should return all notifications', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;

    try {
      const response = getNotificationsAsync(false, 1, token.accessToken);
      const notifications = await response;
      expect('data' in notifications).toBeTruthy();
      expect(notifications.data).toMatchSnapshot();
    } catch (error) {
      const expected = {
        statusCode: 404,
        statusText: 'Not Found',
        errorMessage: [{ key: null, messages: ['Not Found'] }],
      };
      expect(error).toEqual(expected);
    }
  });

  it('should delete notification of user', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;
    const deleteResponse = { data: true };
    try {
      const response = await deleteNotificationAsync(
        1,
        12121212,
        token.accessToken
      );
      expect('data' in response.data).toBeTruthy();
      expect(response.data).toEqual(deleteResponse);
      expect(true).toBeTruthy();
    } catch (error) {
      const expected = {
        statusCode: 404,
        statusText: 'Not Found',
        errorMessage: [{ key: null, messages: ['Not Found'] }],
      };
      expect(error).toEqual(expected);
    }
  });

  it('should return all users notifications', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;

    let filters = {
      limit: 10,
    };

    try {
      const response = getUserNotificationsAsync(filters, 1, token.accessToken);
      const notifications = await response;
      expect('data' in notifications).toBeTruthy();
      expect(notifications.data).toMatchSnapshot();
    } catch (error) {
      expect(error).toHaveProperty('statusCode', 400);
    }
  });

  it('Update notifications', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;

    try {
      await putUserNotification([], token.accessToken);
    } catch (error) {
      expect(error).toHaveProperty('statusCode', 400);
    }
  });

  it('Delete Multiple notifications', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;

    const notificationIds = [];

    try {
      const response = await deleteUserNotificationAsync(
        notificationIds,
        token.accessToken
      );
      expect(response.data).toBeTruthy();
    } catch (error) {
      expect(error).toHaveProperty('statusCode', 400);
    }
  });

  it('should return all counts for notifications', async () => {
    jasmine.DEFAULT_TIMEOUT_INTERVAL = 60000;

    try {
      await getUserNotificationCountAsync(token.accessToken);
    } catch (error) {
      expect(error).toHaveProperty('statusCode', 400);
    }
  });
});
