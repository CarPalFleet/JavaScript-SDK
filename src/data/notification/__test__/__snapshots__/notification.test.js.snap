// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test for retrieving all notifications should return all counts for notifications 1`] = `
Object {
  "read": 0,
  "unread": 0,
}
`;

exports[`Test for retrieving all notifications should return all notifications 1`] = `
Array [
  Object {
    "content": "test",
    "id": "a9f0e92d-6bb0-4126-93cb-2ed8c88cac37",
    "isRead": false,
    "updatedAt": "2018-01-18 07:07:19",
    "userId": 1,
  },
  Object {
    "content": "Test notification",
    "id": "asdf23dsafjak2389",
    "isRead": true,
    "timestamp": "2018-01-01 13:30:00",
    "userId": 1,
  },
]
`;

exports[`Test for retrieving all notifications should return all users notifications 1`] = `Array []`;
